### PR TITLE
dev-python/matplotlib2tikz: Version bump and drop old + dev-python/ImageHash: New package

### PR DESCRIPTION
**Pull request by marbre**
@jlec @SoapGentoo

### CODE REVIEW

File: dev-python/matplotlib2tikz/matplotlib2tikz-0.6.6.ebuild [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 python3_{4,5} )
```
Comment (mgorny) [inline dev-python/matplotlib2tikz/matplotlib2tikz-0.6.6.ebuild]: Any chance to test python3.6 as well?

File: dev-python/matplotlib2tikz/matplotlib2tikz-0.6.6.ebuild [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+SLOT="0"
+IUSE=""
+
+PATCHES=( "${FILESDIR}/${PN}-0.6-init_pipdated.patch" )
+
+RDEPEND="
+	dev-python/matplotlib[${PYTHON_USEDEP}]
+	dev-python/numpy[${PYTHON_USEDEP}]
+	dev-python/pillow[${PYTHON_USEDEP}]"
+DEPEND="${RDEPEND}"
```
Comment (mgorny) [inline dev-python/matplotlib2tikz/matplotlib2tikz-0.6.6.ebuild]: This package seems to provide tests. Why aren't they included in the ebuild?

File: dev-python/matplotlib2tikz/matplotlib2tikz-0.6.6.ebuild (hunk 1 above)
Comment (marbre) [inline dev-python/matplotlib2tikz/matplotlib2tikz-0.6.6.ebuild]: I currently don't have python3.6 installed on my system. I'll try to bump one of my system during the upcoming week. However, matplotlib2tikz is tested with 3.6 by upstream with travis.

File: dev-python/matplotlib2tikz/matplotlib2tikz-0.6.6.ebuild (hunk 2 above)
Comment (marbre) [inline dev-python/matplotlib2tikz/matplotlib2tikz-0.6.6.ebuild]: The sources provided via pypi didn't included the tests. This is the first release getting the sources from github, as the source is no longer directly available on pypi. Therefore we can now implement an python_test() phase. I wont have access to my remote systems before Monday, but I'll give it a try next week.

Comment (marbre) [inline dev-python/matplotlib2tikz/matplotlib2tikz-0.6.6.ebuild]: @mgorny In order to enable testing [imagehash](https://pypi.python.org/pypi/ImageHash) is required, not yet in the tree.

Comment (mgorny) [inline dev-python/matplotlib2tikz/matplotlib2tikz-0.6.6.ebuild]: @marbre, then add it ;-). You can do it in the same PR (but put the extra commit before the bump) or separately, either way works for me. You can add the Python team as a secondary maintainer for that since it seems like a potentially reusable library.

**Comment by mgorny** [comment]
One more note: please remove old versions *after* adding the new one. Otherwise you introduce a state between the two commits where the package is removed. Which also makes it harder to revert properly.

**Comment by marbre** [comment]
@mgorny I reordered my commits and further added ImageHash (Gentoo Bug [616548](https://bugs.gentoo.org/show_bug.cgi?id=616548)).
I currently can't test python3.6, which would not be a problem for matplotlib2tikz (which is tested by upstream with 3.6), but ImageHash doesn't support python3.6 officially. Therefore, I didn't added python3.6 to matplotlib2tikz. Anyway, I think the dependecies for testing matplitlib2tikz can now be fulfilled, but the tests currently fail on my system due to files that do not get installed in the test phasem (some additional images and python scripts, not named `test_*`, see https://github.com/nschloe/matplotlib2tikz/tree/master/test). Still have to figure this out, but unfortunately my time is currently limited.

**Comment by marbre** [comment]
Added the ending newline to the ImageHash ebuild within an interactive rebase.

File: dev-python/ImageHash/ImageHash-3.4.ebuild [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+DESCRIPTION="Image Hashing library"
+HOMEPAGE="https://github.com/JohannesBuchner/imagehash"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+KEYWORDS="~amd64"
+
+LICENSE="BSD-2"
+SLOT="0"
+IUSE="test"
+
+RDEPEND="
```
Comment (mgorny) [inline dev-python/ImageHash/ImageHash-3.4.ebuild]:
```
running test
Searching for scipy
Reading https://pypi.python.org/simple/scipy/
Download error on https://pypi.python.org/simple/scipy/: [Errno -2] Name or service not known -- Some packages may not be found!
Couldn't find index page for 'scipy' (maybe misspelled?)
```

Missing dependency?

File: dev-python/matplotlib2tikz/matplotlib2tikz-0.6.6.ebuild (hunk 1 above)
Comment (mgorny) [inline dev-python/matplotlib2tikz/matplotlib2tikz-0.6.6.ebuild]: Nevermind that. Turns out deps don't support 3.6 anyway, and some of them have broken tests.

File: dev-python/ImageHash/ImageHash-3.4.ebuild (hunk 3 above)
Comment (marbre) [inline dev-python/ImageHash/ImageHash-3.4.ebuild]: Yes, failed to add scipy. Fixed this and further directly bumped matplotlib2tikz to 0.6.7.

**Comment by marbre** [comment]
@mgorny Can we finally merge this or anything else left?

**Comment by mgorny** [comment]
I'm sorry, I'll re-review it today.

File: dev-python/ImageHash/ImageHash-3.4.ebuild [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	dev-python/numpy[${PYTHON_USEDEP}]
+	dev-python/pillow[${PYTHON_USEDEP}]
+	dev-python/pywavelets[${PYTHON_USEDEP}]
+	dev-python/six[${PYTHON_USEDEP}]
+	sci-libs/scipy[${PYTHON_USEDEP}]"
+DEPEND="${RDEPEND}"
```
Comment (mgorny) [inline dev-python/ImageHash/ImageHash-3.4.ebuild]: There's also a setuptools DEPEND missing, and RDEPs are needed at build-time only for tests. But I'll fix that myself.

File: dev-python/matplotlib2tikz/files/matplotlib2tikz-0.6.7-init_pipdated.patch [hunk 5]
```diff
@@ -0,0 +1,13 @@
+Remove version checks using pipdated.
```
Comment (mgorny) [inline dev-python/matplotlib2tikz/files/matplotlib2tikz-0.6.7-init_pipdated.patch]: One more request: please update the patch to remove `pipdated` dependency from setup.py as well. Do that as a separate PR, and no need to hurry -- you can just note that in your TODO for next bump ;-). It doesn't strictly break anything at the moment but the dependency is recorded in the installed package, so setuptools can go weird about it.

**Comment by mgorny** [comment]
Once again, I apologize for the delay. I've tested it thoroughly (which involved a few hours of building and installing deps ;-)), fixed the dependencies and added some test support code. Sadly, tests still fail but I think we're closer. I'd appreciate if you could try locally commenting out the RESTRICT and trying them yourself — maybe I'm missing some flags on matplotlib or so, since I don't have much clue about the package.

**Comment by marbre** [comment]
@mgorny Don't worry about the delay, I think there was enough messed up within my commits. Thanks for all the effort and the assistance :-) I'll modify the patch with the next bump. Further, I think I will get in touch with the package author, who actually was a Gentoo users some years ago and therefore offered help.
